src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -681,7 +681,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
             Dim parameter As IParameterSymbol = boundParameterEqualsValue.Parameter
             Dim value As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() Create(boundParameterEqualsValue.Value))
             Dim kind As OperationKind = OperationKind.ParameterInitializerAtDeclaration
-            Dim isInvalid As Boolean = DirectCast(boundParameterEqualsValue.Value, IOperation).IsInvalid
+            Dim isInvalid As Boolean = value.Value.IsInvalid
```

Comment:
@AlekseyTs basically these 2 cases are the case I missed which assumed BoundNode directly implements IOperation. so the original code used 2 type of nodes interchangeably. the original code cast one to the other in random places and use reference equality to make decision.

all I did here is make sure it normalized to IOperation when it compares reference equality.